client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -240,8 +239,8 @@ const Checkout = React.createClass( {
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
 		} else if ( selectedSite.options.is_domain_only && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainItem = find( cartItems.getAll( cart ), isDomainRegistration );
-			return domainManagementEdit( selectedSite.slug, domainItem.meta );
+			// TODO: Use purchased domain name once it is possible to set it as a primary domain when site is created.
```

Comment:
Indeed, it's asynchronous at the moment (2 minutes) and I think the reason is because we wanted to make sure the domain zone was available. For domain-only-sites we might want to force a primary domain to be set without waiting so users don't see the wordpress.com address